packages/extension/src/view/devtools/app.tsx

```diff
@@ -16,309 +16,64 @@
 /**
  * External dependencies.
  */
-import React, { useRef, useEffect, useCallback, useState } from 'react';
-import { Resizable } from 're-resizable';
+import React, { useRef, useEffect, useState } from 'react';
 import {
-  CookieIcon,
-  CookieIconWhite,
   ExtensionReloadNotification,
-  Sidebar,
-  useSidebar,
-  type SidebarItems,
-  InspectButton,
-  ToastMessage,
+  SidebarProvider,
 } from '@ps-analysis-tool/design-system';
-import { type CookieTableData } from '@ps-analysis-tool/common';
 
 /**
  * Internal dependencies.
  */
 import TABS from './tabs';
-import { useCookie, useSettings } from './stateProviders';
 import './app.css';
-import { Cookies } from './components';
-import useFrameOverlay from './hooks/useFrameOverlay';
+import { Layout } from './components';
 import { getCurrentTabId } from '../../utils/getCurrentTabId';
+import useContextInvalidated from './hooks/useContextInvalidated';
 
 const App: React.FC = () => {
-  const [sidebarWidth, setSidebarWidth] = useState(200);
   const [sidebarData, setSidebarData] = useState(TABS);
   const contextInvalidatedRef = useRef(null);
 
-  const {
-    contextInvalidated,
-    setContextInvalidated,
-    tabFrames,
-    selectedFrame,
-    setSelectedFrame,
-    isInspecting,
-    setIsInspecting,
-    canStartInspecting,
-    tabUrl,
-    frameHasCookies,
-  } = useCookie(({ state, actions }) => ({
-    contextInvalidated: state.contextInvalidated,
-    setContextInvalidated: actions.setContextInvalidated,
-    tabCookies: state.tabCookies,
-    tabFrames: state.tabFrames,
-    selectedFrame: state.selectedFrame,
-    setSelectedFrame: actions.setSelectedFrame,
-    isInspecting: state.isInspecting,
-    setIsInspecting: actions.setIsInspecting,
-    canStartInspecting: state.canStartInspecting,
-    tabUrl: state.tabUrl,
-    frameHasCookies: state.frameHasCookies,
-  }));
-
-  const mainRef = useRef<HTMLElement>(null);
-  const toastMessageRef = useRef<HTMLDivElement>(null);
-
-  const { allowedNumberOfTabs, settingsChanged, handleSettingsChange } =
-    useSettings(({ state, actions }) => ({
-      allowedNumberOfTabs: state.allowedNumberOfTabs,
-      settingsChanged: state.settingsChanged,
-      handleSettingsChange: actions.handleSettingsChange,
-    }));
-
-  const listenToMouseChange = useCallback(() => {
-    if (contextInvalidatedRef.current) {
-      if (!chrome.runtime?.id) {
-        setContextInvalidated(true);
-        localStorage.setItem('contextInvalidated', 'true');
-      }
-    }
-  }, [setContextInvalidated]);
-
-  useEffect(() => {
-    window.addEventListener('mouseover', listenToMouseChange);
-
-    return () => {
-      window.removeEventListener('mouseover', listenToMouseChange);
-    };
-  }, [listenToMouseChange]);
+  const contextInvalidated = useContextInvalidated(contextInvalidatedRef);
```

Comment:
Lets move the context invalidated logic to a custom hook and make App look more cleaner. I moved it in the last commit however haven't tested it. Please take a look.

Reply:
Tested, works as expected 👍